src/interfaces/IMetadataValidator.sol

```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+interface IMetadataValidator {
+    function validate(uint256 userFid, bytes memory key, bytes memory metadata) external returns (bool);
```

Comment:
We're not using it in the validator for _this specific metadata type_, but I'm not sure we want to remove it from the interface. Other, validator types might use it as part of validation. 

The idea with this interface is that the validator has access to the same args as `KeyRegistry.add`. Since there's one validator per keyType/metadataType, those are omitted here—the validator knows which key type and metadata type it's associated with implicitly.